Lanchonete/Lanchonete/ Features/Customization/Presentation/CustomizationViewController.swift

```diff
@@ -57,4 +64,26 @@ class CustomizationViewController: UITableViewController, CustomizationView {
     @IBAction func dismissScreen(_ sender: UIButton) {
         self.dismiss(animated: true, completion: nil)
     }
+    
+    
+    @IBAction func confirmCustomization(_ sender: UIButton) {
+        presenter?.confirmCustomization()
+        self.dismiss(animated: true, completion: nil)
+    }
+}
+extension CustomizationViewController: IngredientCellDelegate {
+    func ingredientCellDidDecreaseQuantity(for index: Int) {
+        confirmButton.isEnabled = presenter!.removeIngredient(from: index)
+        if !confirmButton.isEnabled {
+            confirmButton.alpha = 0.5
+        }
+        reloadData()
+    }
+    
+    func ingredientCellDidIncrementQuantity(for index: Int) {
+        confirmButton.isEnabled = presenter!.addIngredient(from: index)
+        confirmButton.alpha = 1
```

Comment:
As I commented above asking for information from the presenter and taking actions is hard to test (at least you'll have to build expensive UI tests for that). Moving all logic to the presenter makes it easier to write unit tests in the presenter and assert, for example, that when `addIngredient()` runs with success the view was requested to display the button enabled (using a view spy).

For example:
```
func test_WHEN_ingredient_is_successfully_added_THEN_confirm_button_should_be_enabled() {
    // make all arranges to a happy path and inject a view spy that have the same "display(confirm enabled: Bool)" method.
    ...

    // Do the action:
    presenter.addIngredient(from: 0)
    
    // Assert
   XCTAssertEqual(viewSpy.displayConfirmButtonCalled, true)
}

func test_WHEN_ingredient_is_successfully_added_THEN_confirm_button_should_be_enabled() {
    // Now make all arranges to an unhappy path
    ...
    // Do the action:
    presenter.addIngredient(from: 0)
    
    // Assert
   XCTAssertEqual(viewSpy.displayConfirmButtonCalled, false)
}

```

Reply:
Thank you Marcio, I added this tests also